### PR TITLE
Add Java IO Direct jobs: Cdap, HBase, HCatalog, InfluxDb, Kudu

### PR DESCRIPTION
**Pull request by Amar3tto**
The following jenkins jobs have been migrated:

https://ci-beam.apache.org/job/beam_PreCommit_Java_Cdap_IO_Direct_Commit/
https://ci-beam.apache.org/job/beam_PreCommit_Java_Cdap_IO_Direct_Cron/
https://ci-beam.apache.org/job/beam_PreCommit_Java_Cdap_IO_Direct_Phrase/

https://ci-beam.apache.org/job/beam_PreCommit_Java_HBase_IO_Direct_Commit/
https://ci-beam.apache.org/job/beam_PreCommit_Java_HBase_IO_Direct_Cron/
https://ci-beam.apache.org/job/beam_PreCommit_Java_HBase_IO_Direct_Phrase/

https://ci-beam.apache.org/job/beam_PreCommit_Java_HCatalog_IO_Direct_Commit/
https://ci-beam.apache.org/job/beam_PreCommit_Java_HCatalog_IO_Direct_Cron/
https://ci-beam.apache.org/job/beam_PreCommit_Java_HCatalog_IO_Direct_Phrase/

https://ci-beam.apache.org/job/beam_PreCommit_Java_InfluxDb_IO_Direct_Commit/
https://ci-beam.apache.org/job/beam_PreCommit_Java_InfluxDb_IO_Direct_Cron/
https://ci-beam.apache.org/job/beam_PreCommit_Java_InfluxDb_IO_Direct_Phrase/

https://ci-beam.apache.org/job/beam_PreCommit_Java_Kudu_IO_Direct_Commit/
https://ci-beam.apache.org/job/beam_PreCommit_Java_Kudu_IO_Direct_Cron/
https://ci-beam.apache.org/job/beam_PreCommit_Java_Kudu_IO_Direct_Phrase/

### How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

- by creating pull requests with changes in trigger paths
- by adding a comment to the PR
- by making a push

Screenshot from 2023-08-10 11-59-55
Screenshot from 2023-08-10 17-23-01
Screenshot from 2023-08-10 16-59-45
Screenshot from 2023-08-10 16-57-38
Screenshot from 2023-08-10 16-58-34

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+          gradle-command: :sdks:java:io:cdap:build
+          arguments: |
+            -PdisableSpotlessCheck=true \
+            -PdisableCheckStyle=true \
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-code-coverage-report
```
Comment (aleksandr-dudko) [inline .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml:93]: Could it possibly be related to the java-code-coverage-report?

File: .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      matrix:
+        job_name: ["beam_PreCommit_Java_Cdap_IO_Direct"]
+        job_phrase: ["Run Java_Cdap_IO_Direct PreCommit"]
+    timeout-minutes: 60
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
+      github.event_name == 'schedule' ||
+      github.event_name == 'workflow_dispatch' ||
+      github.event.comment.body == 'Run Java_Cdap_IO_Direct PreCommit'
```
Comment (aleksandr-dudko) [inline .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml:73]: Could it possibly be related to the startswith(github.event.comment.body,'Run Java_Cdap_IO_Direct PreCommit')

Comment (Amar3tto) [reply .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml:73]: No, I think here will be always the same comment phrase

File: .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml (hunk 1 above)
Comment (Amar3tto) [reply .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml]: Fixed

File: .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml (hunk 2 above)
Comment (aleksandr-dudko) [reply .github/workflows/beam_PreCommit_Java_Cdap_IO_Direct.yml:73]: I'm not sure about the answer, but I'm curious if it's possible to use the value of the matrix element here?
github.event.comment.body == matrix.job_phrase

**Pull request merged**